### PR TITLE
Re-enable E2E proxy test

### PR DESCRIPTION
**Pull request by massand**
- Re-enable E2E pipeline job to run tests in proxy environment
- Fixed some tests to run using Websockets transport (which is generally the case for E2E tests)
- Improve UploadSupportBundle test reliability in self-host environment by reducing requested log time range
- Fixed typo in traces within UploadSupportBundle product code
- Filtered out Nested Edge tests and temporarily filtered out tests that timeout in proxy environment

### CODE REVIEW

**Review by gordonwang0 — DISMISSED** [review]
Please run the pipeline on this PR to make sure it works before merging.

File: builds/e2e/templates/e2e-run.yaml [hunk 1]
```diff
@@ -38,9 +38,20 @@ steps:
     {
       $filter = 'Category=nestededge_isa95'
     }
+    elseif ($test_type -eq 'http_proxy')
```
Comment (gordonwang0) [inline builds/e2e/templates/e2e-run.yaml:41]: I think it's better to create a new group for tests to disable behind the proxy instead of disabling tests by name individually.

File: test/Microsoft.Azure.Devices.Edge.Test/IoTEdgeCheck.cs [hunk 2]
```diff
@@ -100,11 +100,15 @@ public async Task IoTEdge_check()
...
 
-            // TODO: iotedge check currently has 2 legitimate errors in it. When we update the aziot-submodule, the errors
-            // will go away, and we should change this assertion to empty string instead of passing on 2 errors.
-            // This is better than disabling the test, because it still tests iotedge check to some extent, and it will
-            // remind us with a failed run to update this test.
-            Assert.AreEqual(string.Empty, errors_number);
+            // When test runs behind proxy, MQTT and AMQP checks legitimately fail in three different connectivity checks
+            if (Context.Current.EdgeProxy.HasValue)
+            {
+                Assert.AreEqual("6 check(s) raised errors.", errors_number);
```
Comment (gordonwang0) [inline test/Microsoft.Azure.Devices.Edge.Test/IoTEdgeCheck.cs:106]: This is kinda hacky. I think it's better to use the `--dont-run` option to `iotedge check` and disable the checks that shouldn't run behind a proxy there.

File: test/Microsoft.Azure.Devices.Edge.Test/PlugAndPlay.cs [hunk 3]
```diff
@@ -29,10 +29,10 @@ public class PlugAndPlay : SasManualProvisioningFixture
         const string TestModelId = "dtmi:edgeE2ETest:TestCapabilityModel;1";
         const string LoadGenModuleName = "loadGenModule";
 
-        [TestCase(Protocol.Mqtt, false)]
-        [TestCase(Protocol.Amqp, false)]
-        [TestCase(Protocol.Mqtt, true)]
-        [TestCase(Protocol.Amqp, true)]
+        [TestCase(Protocol.MqttWs, false)]
```
Comment (gordonwang0) [inline test/Microsoft.Azure.Devices.Edge.Test/PlugAndPlay.cs:32]: Why were these changed?

Comment (massand) [reply test/Microsoft.Azure.Devices.Edge.Test/PlugAndPlay.cs:32]: Mentioned this in the description- Fixed some tests to run using Websockets transport (which is generally the case for E2E tests). If this assumption is incorrect, I can revert this. But, this should work for the normal (non-proxy) tests the same.

File: builds/e2e/templates/e2e-run.yaml (hunk 1 above)
Comment (massand) [reply builds/e2e/templates/e2e-run.yaml:41]: Agreed on categorizing tests. I am still investigating some tests before adding test categories to filter them out.

File: test/Microsoft.Azure.Devices.Edge.Test/IoTEdgeCheck.cs (hunk 2 above)
Comment (massand) [reply test/Microsoft.Azure.Devices.Edge.Test/IoTEdgeCheck.cs]: Fixed

**Review by gordonwang0 — APPROVED** [review]
